robots/cmd/test-label-analyzer/cmd/filter/matching-tests.go

```diff
@@ -0,0 +1,138 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ */
+
+package filter
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"kubevirt.io/project-infra/robots/pkg/git"
+	testlabelanalyzer "kubevirt.io/project-infra/robots/pkg/test-label-analyzer"
+	"os"
+	"strings"
+)
+
+type filterStatsMatchingTestsOptions struct {
+	outputFilePath string
+
+	outputFile   *os.File
+	inputVersion string
+}
+
+func (o *filterStatsMatchingTestsOptions) validate() error {
+	if o.outputFilePath == "" {
+		return fmt.Errorf("output-file needs to be provided")
+	}
+	_, err := os.Stat(o.outputFilePath)
+	if !os.IsNotExist(err) {
+		return fmt.Errorf("output file %q exists", o.outputFilePath)
+	}
+	o.outputFile, err = os.Create(o.outputFilePath)
+	if err != nil {
+		return fmt.Errorf("could not create output file %q: %v", o.outputFilePath, err)
```

Comment:
Maybe we can change %v to %w here to wrap the underlying error, providing more context to the caller?

Reply:
There are other places that it can be applied as well

Reply:
Thanks, TIL :+1: